cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -370,9 +370,18 @@ var ControllersDisabledByDefault = sets.NewString(
 	"tokencleaner",
 )
 
-const (
-	saTokenControllerName = "serviceaccount-token"
-)
+// NewOrderedControllerInitializers returns an order list of named controller groups (you can start more than one in an init func) paired to their InitFunc.
+// Note that the controller will be run in the order defined in the list.
+func NewOrderedControllerInitializers(rootClientBuilder controller.ControllerClientBuilder) []orderedControllerContext {
+	return []orderedControllerContext{
+		// Always start the SA token controller first using a full-power client, since it needs to mint tokens for the rest
+		// If this fails, just fail since other controllers won't be able to get credentials.
+		{"serviceaccount-token", wrapStartServiceAccountTokenController(rootClientBuilder), true},
+
+		// Node Lifecycle controller is important because it's responsible for noticing and responding when nodes go down.
```

Comment:
Just being important doesn't provide detail about why this must be started before the others.  Why is this controller more equal than others?

Reply:
added a comment, I hope it's okay now.

Reply:
> added a comment, I hope it's okay now.

I'm still not convinced by this.  It can be started in an unordered manner after the service account tokens are created. Service account tokens are first because without them the rest will never start.